Essential corrections to R4 standard swagger for GA

leonardbf opened the pull request:
Some missing required fields for capacityPool and volume.
Some missing attributes in volume properties (essential for R4 usage in
swagger generated SDKs and command tools).
Name correction of resource type for availability check.
Correction to default API version.

These changes are to the stable 2019-05-01 version of the API, created for GA. Some of these look to be breaking changes. However GA has not yet been announced and these changes are necessary to generate full command set in the CLI and PowerShell.




### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

leonardbf committed:
NFSAAS-2436 corrections to R4 standard

Some missing required fields for capacityPool and volume.
Some missing attributes in volume properties (essential for R4 usage in
swagger generated SDKs and command tools).
Correction to default API version.

leonardbf committed:
Merge pull request #25 from leonardbf/NFSAAS-2436-R4-corrections

NFSAAS-2436 corrections to R4 standard

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-js

AutorestCI commented:
# Automation for azure-sdk-for-go

azuresdkci commented:
Can one of the admins verify this patch?

adxsdknet commented:
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/6400 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/6400/commits/5888d873a0198d48067cd406fbd132d222cdbafd

leonardbf committed:
NFSAAS-2436 corrections to R4 standard

Some missing required fields for capacityPool and volume.
Correction to availability function resource names.
Some missing attributes in volume properties (essential for R4 usage in
swagger generated SDKs and command tools).

leonardbf committed:
Merge pull request #26 from leonardbf/NFSAAS-2436-R4-corrections

NFSAAS-2436 corrections to R4 standard

AutorestCI commented:
# Automation for azure-sdk-for-java

leonardbf committed:
NFSAAS-2436 corrections to R4 standard

Some missing required fields for capacityPool and volume.
Corrections to availability function resource type names.
Some missing attributes in volume properties (essential for R4 usage in swagger generated SDKs and command tools).
Correction to default API version.

leonardbf committed:
Merge pull request #27 from leonardbf/NFSAAS-2436-R4-corrections

NFSAAS-2436 corrections to R4 standard

leonardbf committed:
NFSAAS-2436 corrections to R4 standard

Some missing required fields for capacityPool and volume.
Corrections to availability function resource type  names.
Some missing attributes in volume properties (essential for R4 usage in swagger generated SDKs and command tools).
Correction to default API version.

leonardbf committed:
Merge pull request #28 from leonardbf/NFSAAS-2436-R4-corrections

NFSAAS-2436 corrections to R4 standard

leonardbf commented:
@praries880 Hi, I see you have been allocated for review. I see two failures in the pipeline. One does not look like it comes from my change. The other is puzzling because it says not valid json. However the part being reported passes local validation and anyway has not changed since the last time this swagger was accepted and passed. Can you advise on this please when you review. Thanks.

leonardbf commented:
@praries880 Hi, any info on the comment above about failures or chance to otherwise review this PR? Thanks.

praries880 commented:
the error in the breaking change check tool seems to be a bug with the tool : 
https://github.com/Azure/openapi-diff/issues/150

praries880 commented on specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-05-01/netapp.json:
```diff
@@ -1372,6 +1372,10 @@
     "poolProperties": {
       "description": "Pool properties",
       "type": "object",
+      "required": [
```
@leonardbf  This seems to be a breaking change... Is this model used for a request?

leonardbf replied:
> @leonardbf This seems to be a breaking change... Is this model used for a request?

@praries880 yes this is used for requests, pool creation for instance. The fact is that the required field here was missing, i.e. the swagger did not correctly represent the resource provider. This was therefore a correction.

leonardbf commented:
Any news on whether these changes can be merged? Thanks.

KrisBash reviewed (approved):
Seems fine per ARM review, given explanation on unshipped state

leonardbf commented:
Thanks for the approval. Any chance of being merged?

praries880 commented:
@sergey-shandar  can you kindly merge this PR?
